ulohy_04/Caesar_cipher.py

```diff
@@ -0,0 +1,25 @@
+# Vstup - ošetření, že text bude jen ASCII
+# a klíč jen celé a kladné číslo do 26 (máme 26 znaků)
+while True:
+    text = input("Zadej text, který chceš zakódovat: ")
+    if text.isascii():
+        while True:
+            key = input("Jaký klíč chceš použít? ")
+            if key.isdigit() and int(key) > 0 and int(key) % 26 != 0:
+                break
+            else:
+                print('''Klíč musí být celé kladné číslo.
+Násobek 26 nic nezašifruje.''')
+        break
+    print("Bohužel text obsahuje nepovolené znaky, které nejsou ASCII.")
+# Šifra - velká písmena 65 - 90, malá písmena 97 - 122 (26 písmen)
+new_text = ""
+for i in range(len(text)):
+    char = text[i]
+    if (char.isupper()):
+        new_text += chr((ord(char) + int(key) - 65) % 26 + 65)
```

Comment:
Funkčně dobře i jsi to nahoře pěkně okomentovala, co to je za čísla. Napadá tě přesto, jak to vylepšit? :)

Reply:
Jediné řešení, co mě napadá je nahradit číslo 65 za `ord("A")` a číslo 97 nahradit `ord("a")`, ale netuším, zda je to řešení, které máš na mysli.

Reply:
Jo, jdeš na to dobře. Ale takto by se pak musel `ord('A')` vyhodnocovat při každém písmenu. Takže by bylo lepší si toto nadefinovat vhodně pojmenovanou konstantu ještě před hlavním cyklem. Ale je to drobnost.